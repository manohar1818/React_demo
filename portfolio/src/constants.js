export const HOME="Home";
export const ABOUT="About";
export const CONTACT="Contact";
export const GALLERY="Gallery";
export const NAME="Welcome All , I'm Manohar Reddy";
export const NAME_LABEL="Name";
export const EMAIL_LABEL="Email";
export const MESSAGE_LABEL="Message";
export const GALLERY_ADMIN="GalleryAdmin";
export const URL_LABEL="Url";
export const ABOUT_ME=" I am Manohar Reddy from Mahatma Gandhi Institute of Technology. I usually love learning new Technologies . My Hobies playing Badminton ";
export const TECHNICAL_SKILLS="Technical Skills";
export const JAVA="Java";
export const DBMS="DBMS";
export const SPRING="Spring";
export const SPRING_BOOT="Spring boot";
export const REST="REST";
export const NETWORKING="Networking";
export const LINUX="Linux";
export const HTML_CSS="HTML,CSS";
export const PROJECTS="Projects";
export const GATE_PASS_MANAGEMENT_SYSTEM="Gate Pass Management System";
export const MAKING_DATA_BASED_DESCIONS_USING_GOOGLE_ANALYTICS="Making Data Based Decisions Using Google Analytics";
export const ONLINE_DOCTOR_APPOINTMENT="Online Doctor Appointment";
export const SPRING_BOOT_REST_API_MOVIES="Spring boot REST API which gives information about movies and associated actors";
export const COURSES_PROJECT="Spring boot application which gives information about courses and related topics";
export const DETECTION_PREDICTION_CLOUDS="Detection and Prediction of Clouds";
export const REACT="React";
export const CONNECT="connect with me";
export const SUBMIT="submit";
export const ADD="Add";
export const ADD_IMAGE="Add Image";
export const LOGOUT="Logout";
export const DESIGNATION="Full Stack Developer Intern";
export const SAVE="save";
export const LOGIN="login";
export const ERROR_PAGE="Error Page";
export const LOGIN_FIRST="You have to login first";
export const BACK_TO_LOGIN_PAGE="Back to login page";
export const LOGIN_PAGE="LoginPage";
export const INVALID_NAME="invalid name";
export const INVALID_EMAIL="invalid email";
export const INVALID_MESSAGE="invalid message";
export const INVALID_URL="invalid url";
export const LOADING="Loding";
