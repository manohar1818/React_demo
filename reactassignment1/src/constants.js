export const HOME="Home";
export const ABOUT="About";
export const CONTACT="Contact";
export const GALLERY="Gallery";
export const NAME="Welcome All, I'm Manohar Reddy";
export const NAME_LABEL="Name";
export const EMAIL_LABEL="Email";
export const MESSAGE_LABEL="Message";
export const GALLERY_ADMIN="GalleryAdmin";
export const URL_LABEL="Url";
export const ABOUT_ME=" I am Manohar Reddy from Mahatma Gandhi Institute of Technology.I'm passionate about coding and learning new technologies.In free time I like to chill at home watching movies and colouring .I love to be surrounded by positive and motivating people. ";
export const TECHNICAL_SKILLS="Technical Skills";
export const PERSONAL="Personal details";
export const MYNAME = 'Name : Gangidi Manohar Reddy';
export const CONTACTNO = "Contact-no : 7995520569";
export const EMAIL = "Email : gmr1818@gmail.com";
export const DOB = "DOB : 18 th June 1998";
export const LANGUAGES = "Languages : English , Hindi , Telugu";
export const HOBBIES ="Hobbies : Real Estate , Badminton";
export const JAVA="Java";
export const DBMS="DBMS";
export const SPRING="Python";
export const SPRING_BOOT="Spring boot";
export const REST="Machine Learning";
export const NETWORKING="Networking";
export const LINUX="Linux";
export const HTML_CSS="HTML,CSS";
export const PROJECTS="Projects";
export const a="Crop yield usig ML";
export const b="College Management System";
export const c="Sports App";
export const d="Smart Parking System";
export const REACT="React";
export const CONNECT="connect with me";
export const SUBMIT="submit";
export const ADD="Add";
export const ADD_IMAGE="Add Image";
export const LOGOUT="Logout";
export const DESIGNATION="Intern at Zemoso Technologies";
export const SAVE="save";
export const LOGIN="login";
export const ERROR_PAGE="Error Page";
export const LOGIN_FIRST="You have to login first";
export const BACK_TO_LOGIN_PAGE="Back to login page";
export const LOGIN_PAGE="LoginPage";
export const INVALID_NAME="invalid name";
export const INVALID_EMAIL="invalid email";
export const INVALID_MESSAGE="invalid message";
export const INVALID_URL="invalid url";
export const LOADING="Loding";
