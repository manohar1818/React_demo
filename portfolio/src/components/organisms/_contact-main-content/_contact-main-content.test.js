import React from 'react';
import { create } from "react-test-renderer";
import { render ,fireEvent,screen} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ContactMainContent from './_contact-main-content';
import * as Constants from "../../../constants";
describe("ContactMainContent component",() =>{
    test("Matches the snapshot",() => {
        const contact= create(<ContactMainContent/>);
        expect(contact.toJSON()).toMatchSnapshot();

    })
    test("ContactMainContent is rendered",() => {

        const { getByTestId, getByText } = render(<ContactMainContent/>);
        expect(getByTestId("contact")).toBeInTheDocument;
        
    });

    test("initially no error messages", () => {
        const { getByRole ,getByTestId} = render(<ContactMainContent />);
       
        expect(() => screen.getByText(Constants.INVALID_NAME)).toThrow('Unable to find an element');
        expect(() => screen.getByText(Constants.INVALID_EMAIL)).toThrow('Unable to find an element');
        expect(() => screen.getByText(Constants.INVALID_MESSAGE)).toThrow('Unable to find an element');
       
      });
    
      test("error msgs for inavlid inputs", () => {
        const {getByTestId} = render(<ContactMainContent />);
        fireEvent.change(getByTestId("name"), {
            target: { value: " ram" },
          });
          expect(getByTestId("emailError")).toBeInTheDocument();
          fireEvent.change(getByTestId("email"), {
            target: { value: "ram12@" },
          });
          expect(getByTestId("nameError")).toBeInTheDocument();
          fireEvent.change(getByTestId("message"), {
            target: { value: " " },
          });
          expect(screen.getByText(Constants.INVALID_NAME)).toBeInTheDocument();
          expect(screen.getByText(Constants.INVALID_EMAIL)).toBeInTheDocument();
          //expect(screen.getByText(Constants.INVALID_MESSAGE)).toBeInTheDocument();
       
     });
    
     test("valid inputs", () => {
        const { getByTestId} = render(<ContactMainContent />);
        fireEvent.change(getByTestId("name"), {
            target: { value: "Ram" },
          });
          expect(getByTestId("emailError")).toBeInTheDocument();
          fireEvent.change(getByTestId("email"), {
            target: { value: "ram12@gmail.com" },
          });
          expect(getByTestId("nameError")).toBeInTheDocument();
          fireEvent.change(getByTestId("message"), {
            target: { value: "hii" },
          });
          expect(() => screen.getByText(Constants.INVALID_NAME)).toThrow('Unable to find an element');
          expect(() => screen.getByText(Constants.INVALID_EMAIL)).toThrow('Unable to find an element');
          expect(() => screen.getByText(Constants.INVALID_MESSAGE)).toThrow('Unable to find an element');
     });

});