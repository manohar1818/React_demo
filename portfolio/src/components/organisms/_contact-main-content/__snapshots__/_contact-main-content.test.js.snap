// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactMainContent component Matches the snapshot 1`] = `
<div
  className="makeStyles-styleContactMainContent-1"
  data-testid="contact"
>
  <h3
    className="MuiTypography-root makeStyles-heading-4 MuiTypography-h3"
  >
    connect with me
  </h3>
  <div
    className="makeStyles-forms-2"
  >
    <form>
      <div
        className="makeStyles-input-3"
      >
        <div
          data-testid="input"
        >
          <h5
            className="MuiTypography-root MuiTypography-h5"
          >
            Name
          </h5>
          <div
            className="makeStyles-styleInputFields-6"
            data-testid="inputField"
          >
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline"
              data-testid="inputAtom"
              onClick={[Function]}
              required={true}
              style={
                Object {
                  "color": "#000",
                }
              }
            >
              <input
                className="MuiInputBase-input MuiInput-input"
                data-testid="name"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
              />
            </div>
          </div>
        </div>
      </div>
      <p
        className="MuiTypography-root makeStyles-error-5 MuiTypography-body1"
        data-testid="nameError"
      />
      <div
        className="makeStyles-input-3"
      >
        <div
          data-testid="input"
        >
          <h5
            className="MuiTypography-root MuiTypography-h5"
          >
            Email
          </h5>
          <div
            className="makeStyles-styleInputFields-6"
            data-testid="inputField"
          >
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline"
              data-testid="inputAtom"
              onClick={[Function]}
              required={true}
              style={
                Object {
                  "color": "#000",
                }
              }
            >
              <input
                className="MuiInputBase-input MuiInput-input"
                data-testid="email"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
              />
            </div>
          </div>
        </div>
      </div>
      <p
        className="MuiTypography-root makeStyles-error-5 MuiTypography-body1"
        data-testid="emailError"
      />
      <div
        className="makeStyles-input-3"
      >
        <div
          data-testid="input"
        >
          <h5
            className="MuiTypography-root MuiTypography-h5"
          >
            Message
          </h5>
          <div
            className="makeStyles-styleInputFields-6"
            data-testid="inputField"
          >
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline"
              data-testid="inputAtom"
              onClick={[Function]}
              required={true}
              style={
                Object {
                  "color": "#000",
                }
              }
            >
              <input
                className="MuiInputBase-input MuiInput-input"
                data-testid="message"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
              />
            </div>
          </div>
        </div>
      </div>
      <p
        className="MuiTypography-root makeStyles-error-5 MuiTypography-body1"
        data-testid="messageError"
      />
      <div
        className="makeStyles-button-7"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
          data-testid="button"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <span
              className="MuiTypography-root MuiTypography-button"
            >
               
              submit
            </span>
          </span>
        </button>
      </div>
    </form>
  </div>
</div>
`;
